### PR TITLE
Add compatibility for ROS 2 user info API

### PR DESCRIPTION
**Pull request by austinzheng**
Changes:
- Fixed some issues caused by latest ROS in tests
- Rewrote the user info models to reflect how ROS 2 sends down that data
- Re-enabled unit test

TODO:
- [ ] ~Re-enable the Objective-C version of this test~
- [x] Changelog entry
- [x] Retitle the commit to be more descriptive

### CODE REVIEW

**Comment by austinzheng** [comment]
I think this can be reviewed; I'll make more PRs as more tests are re-enabled.

File: Realm/ObjectServerTests/RLMObjectServerTests.mm [hunk 1]
```diff
@@ -398,8 +397,8 @@ - (void)disabled_testRetrieveUserInfo {
                                completion:^(RLMSyncUserInfo *info, NSError *err) {
                                    XCTAssertNil(err);
                                    XCTAssertNotNil(info);
-                                   XCTAssertEqualObjects(info.provider, RLMIdentityProviderUsernamePassword);
-                                   XCTAssertEqualObjects(info.providerUserIdentity, nonAdminUsername);
+                                   XCTAssertTrue([info.accounts count] > 0);
```
Comment (tgoyne) [inline Realm/ObjectServerTests/RLMObjectServerTests.mm]: `XCTAssertGreaterThan()`?

File: Realm/RLMJSONModels.h [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 #import <Foundation/Foundation.h>
 
+#import "RLMSyncUser.h"
```
Comment (tgoyne) [inline Realm/RLMJSONModels.h]: A forward-declaration of `RLMSyncUserAccountInfo` should suffice here.

File: Realm/RLMSyncUser.h [hunk 3]
```diff
@@ -276,46 +277,60 @@ NS_SWIFT_UNAVAILABLE("Use the full version of this API.");
...
 
 /**
  The identity issued to this user by the Realm Object Server.
  */
 @property (nonatomic, readonly) NSString *identity;
 
 /**
+ Metadata about this user stored on the Realm Object Server.
+ */
+@property (nonatomic, readonly) NSDictionary *metadata;
```
Comment (tgoyne) [inline Realm/RLMSyncUser.h]: `NSDictionary<NSString *, NSString *>`?

File: Realm/RLMSyncUser.h [hunk 4]
```diff
@@ -276,46 +277,60 @@ NS_SWIFT_UNAVAILABLE("Use the full version of this API.");
...
- A data object representing information about a user that was retrieved from a user lookup call.
+ A data object representing a user account associated with a user.
+
+ @see `RLMSyncUserInfo`
  */
-@interface RLMSyncUserInfo : NSObject
+@interface RLMSyncUserAccountInfo : NSObject
+
+/// The authentication provider which manages this user account.
+@property (nonatomic, readonly) NSString *provider;
```
Comment (bdash) [inline Realm/RLMSyncUser.h]: Is there a reason this is a string rather than `RLMIdentityProvider`?

**Comment by austinzheng** [comment]
The CI checks that failed all failed because they timed out.

**Pull request merged**